src/targets/__tests__/awsLambda.test.ts

```diff
@@ -0,0 +1,216 @@
+    }
+    expect(getAwsLayerName()).toBe(defaultLayerName);
+  });
+
+  test('custom environment variable', () => {
+    const customEnvName = 'test-env-layer-name';
+    process.env.AWS_LAYER_NAME = customEnvName;
+    expect(getAwsLayerName()).toBe(customEnvName);
+  });
+});
+
+describe('get aws config environment variables', () => {
+  const oldEnvVariables = process.env;
+
+  beforeEach(() => {
+    jest.resetModules(); // Clear the cache.
+    process.env = { ...oldEnvVariables }; // Restore environment
+  });
+
+  afterAll(() => {
+    process.env = { ...oldEnvVariables }; // Restore environment
+  });
+
+  function deleteTargetOptionsFromEnvironment() {
+    if ('AWS_ACCESS_KEY_ID' in process.env) {
+      delete process.env.AWS_ACCESS_KEY_ID;
+    }
+    if ('AWS_SECRET_ACCESES_KEY' in process.env) {
+      delete process.env.AWS_SECRET_ACCESES_KEY;
+    }
+  }
+
+  test('errors on missing environment variables', () => {
+    deleteTargetOptionsFromEnvironment();
+    try {
+      getAwsLambdaTarget();
+    } catch (e) {
+      expect(e instanceof ConfigurationError).toBe(true);
+    }
+  });
+
+  test('success on environment variables', () => {
+    deleteTargetOptionsFromEnvironment();
+    setAwsEnvironmentVariables();
+    // AwsLambdaTarget needs the environment variables to initialize.
+    getAwsLambdaTarget();
+  });
+});
+
+describe('publish', () => {
+  beforeAll(() => {
+    setAwsEnvironmentVariables();
+  });
+
+  const noArtifactsForRevision = jest.fn().mockImplementation(function() {
+    return [];
+  });
+
+  test('error on missing zip file', async () => {
+    const awsTarget = getAwsLambdaTarget();
+    awsTarget.getArtifactsForRevision = noArtifactsForRevision.bind(
+      AwsLambdaTarget
+    );
+    // `publish` should report an error. When it's not dry run, the error is
+    // thrown; when it's on dry run, the error is logged and `undefined` is
+    // returned. Thus, both alternatives have been considered.
+    try {
+      const noPackageFound = await awsTarget.publish('version', 'revision');
+      expect(noPackageFound).toBe(undefined);
+    } catch (error) {
+      expect(error instanceof Error).toBe(true);
+    }
+  });
+
+  const twoArtifactsForRevision = jest.fn().mockImplementation(function() {
+    return ['file1', 'file2'];
+  });
+
+  test('error on having too many files', async () => {
+    const awsTarget = getAwsLambdaTarget();
+    awsTarget.getArtifactsForRevision = twoArtifactsForRevision.bind(
+      AwsLambdaTarget
+    );
+    // `publish` should report an error. When it's not dry run, the error is
+    // thrown; when it's on dry run, the error is logged and `undefined` is
+    // returned. Thus, both alternatives have been considered.
+    try {
+      const tooManyPackagesFound = await awsTarget.publish(
+        'version',
+        'revision'
+      );
+      expect(tooManyPackagesFound).toBe(undefined);
+    } catch (error) {
+      expect(error instanceof Error).toBe(true);
```

Comment:
This test looks suspiciously similar to the previous one. Wdyt about actually checking the error message to confirm that it fails for the right reason?